### PR TITLE
Move get/setindex and readavailable out of helpdb

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by KristofferC — APPROVED** [review]
Superficially, LGTM

File: base/abstractarray.jl [hunk 1]
```diff
@@ -873,7 +873,35 @@ end
 #       getindex(::A{T,N}, ::Vararg{Int, N}) where {T,N} # if IndexCartesian()
 # If the subtype hasn't defined the required method, it falls back to the
 # _getindex function again where an error is thrown to prevent stack overflows.
+"""
+    getindex(A, inds...)
+
+Returns a subset of array `A` as specified by `inds`, where each `ind` may be an
```
Comment (tkelman) [inline base/abstractarray.jl]: Return

File: base/array.jl [hunk 2]
```diff
@@ -190,6 +190,27 @@ similar(a::Array, T::Type, dims::Dims{N}) where {N} = Array{T,N}(dims)
 similar(a::Array{T}, dims::Dims{N}) where {T,N}     = Array{T,N}(dims)
 
 # T[x...] constructs Array{T,1}
+"""
+    getindex(type[, elements...])
+
+Construct a 1-d array of the specified type. This is usually called with the syntax
+`Type[]`. Element values can be specified using `Type[a,b,c,...]`.
+
+# Example
```
Comment (tkelman) [inline base/array.jl]: might be switching all of these to plural

**Comment by kshyatt** [comment]
Ready to roll?

**Pull request merged**